src/tests/ftest/server/daos_server_restart.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2020 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from __future__ import print_function
+
+from avocado import fail_on
+from apricot import TestWithServers
+from daos_utils import DaosCommand
+from command_utils import CommandFailure
+
+
+class DaosServerTest(TestWithServers):
+    """Daos server tests.
+
+    Test Class Description:
+        Test to verify that the daos_io_server, daos_server starts/stops
+        and reformat.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a DaosServerTest object."""
+        super(DaosServerTest, self).__init__(*args, **kwargs)
+
+    @fail_on(CommandFailure)
```

Comment:
You should add another `@fail_on()` decorator with the ServerFailed exception:
`@fail_on(ServerFailed)` otherwise, the test will not fail when an issue is encountered with the server commands.

Reply:
Thanks for the comment, please see the new update.